### PR TITLE
wush 0.1.2 (new formula)

### PR DESCRIPTION
**Pull request by indianets**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/w/wush.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Wush < Formula
+  desc "Simplest & fastest way to transfer files between computers via wireguard"
```
Comment (carlocab) [inline Formula/w/wush.rb:2]:
```suggestion
  desc "Transfer files between computers via WireGuard"
```

File: Formula/w/wush.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  license "CC0-1.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w -X main.version=#{version}"), "./cmd/..."
+  end
+
+  test do
+    system bin/"wush", "version"
```
Comment (carlocab) [inline Formula/w/wush.rb:15]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Comment (indianets) [reply Formula/w/wush.rb:15]: I'll try to implement that, but commands with wush run until killed, so I referred sshfs for the test case. I need an example formula which implements more test cases while running never ending commands.

Comment (SMillerDev) [reply Formula/w/wush.rb:15]: How about httpd? That runs a (never ending) server

Comment (indianets) [reply Formula/w/wush.rb:15]: Yes, I did something similar, thanks for suggesting. I am in process of enhancing that a little more.

**Comment by chenrui333** [comment]
need commit squash

File: Formula/w/wush.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  desc "Transfer files between computers via WireGuard"
+  homepage "https://github.com/coder/wush"
+  url "https://github.com/coder/wush/archive/refs/tags/v0.1.2.tar.gz"
+  sha256 "8e1ffd24b51775ac68116b381cf4523a54d4e7742e440b7d279418faa52904eb"
+  license "CC0-1.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w -X main.version=#{version}"), "./cmd/..."
```
Comment (chenrui333) [inline Formula/w/wush.rb:11]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w -X main.version=#{version}"), "./cmd/wush"
```

**Comment by indianets** [comment]
> need commit squash

Done, also thank you for your other suggestions and good work for community.

**Review by chenrui333 — APPROVED** [review]

**Review by stefanb — APPROVED** [review]

**Pull request merged**